src/raw_node.rs

```diff
@@ -174,6 +152,68 @@ impl Ready {
         }
         rd
     }
+
+    /// The current volatile state of a Node.
+    /// SoftState will be nil if there is no update.
+    /// It is not required to consume or store SoftState.
+    #[inline]
+    pub fn ss(&self) -> Option<&SoftState> {
+        self.ss.as_ref()
+    }
+
+    /// The current state of a Node to be saved to stable storage BEFORE
+    /// Messages are sent.
+    /// HardState will be equal to empty state if there is no update.
+    #[inline]
+    pub fn hs(&self) -> Option<&HardState> {
+        self.hs.as_ref()
+    }
+
+    /// States can be used for node to serve linearizable read requests locally
+    /// when its applied index is greater than the index in ReadState.
+    /// Note that the read_state will be returned when raft receives MsgReadIndex.
+    /// The returned is only valid for the request that requested to read.
+    #[inline]
+    pub fn read_states(&self) -> &[ReadState] {
+        &self.read_states
+    }
+
+    /// Entries specifies entries to be saved to stable storage BEFORE
+    /// Messages are sent.
+    #[inline]
+    pub fn entries(&self) -> &[Entry] {
+        &self.entries
+    }
+
+    /// Snapshot specifies the snapshot to be saved to stable storage.
+    #[inline]
+    pub fn snapshot(&self) -> &Snapshot {
+        &self.snapshot
+    }
+
+    /// CommittedEntries specifies entries to be committed to a
+    /// store/state-machine. These have previously been committed to stable
+    /// store.
+    #[inline]
+    pub fn committed_entries(&self) -> Option<&[Entry]> {
```

Comment:
So users have to clone the memory to handle committed entries in other thread now?